x/ecocredit/server/basket/take_test.go

```diff
@@ -0,0 +1,263 @@
+package basket_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/store"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+
+	"github.com/regen-network/regen-ledger/types/math"
+
+	"github.com/cosmos/cosmos-sdk/orm/model/ormdb"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormtable"
+	"github.com/cosmos/cosmos-sdk/orm/testing/ormtest"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/golang/mock/gomock"
+	basketv1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	baskettypes "github.com/regen-network/regen-ledger/x/ecocredit/basket"
+	"github.com/regen-network/regen-ledger/x/ecocredit/server"
+	"github.com/regen-network/regen-ledger/x/ecocredit/server/basket"
+	"github.com/regen-network/regen-ledger/x/ecocredit/server/basket/mocks"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gotest.tools/v3/assert"
+)
+
+type suite struct {
+	t               *testing.T
+	db              ormdb.ModuleDB
+	stateStore      basketv1.StateStore
+	ctx             context.Context
+	k               basket.Keeper
+	ctrl            *gomock.Controller
+	acct            sdk.AccAddress
+	bankKeeper      *mocks.MockBankKeeper
+	ecocreditKeeper *mocks.MockEcocreditKeeper
+	fooBasketId     uint64
+	barBasketId     uint64
+	storeKey        *sdk.KVStoreKey
+	sdkCtx          sdk.Context
+}
```

Comment:
why not using https://pkg.go.dev/github.com/stretchr/testify/suite ?

Reply:
Using gotest with orm and got used to it. It supported protobuf better